CLI/counter.c

```diff
@@ -0,0 +1,398 @@
+        free(data);
+    }
+
+    return NO_ERROR;
+}
+
+static int build_json_counter_entry(json_t *parent, psabpf_counter_context_t *ctx, psabpf_counter_entry_t *entry)
+{
+    if (parent == NULL)
+        return ENOMEM;
+
+    json_t *json_key = json_object();
+    if (json_key == NULL)
+        return ENOMEM;
+    json_object_set(parent, "key", json_key);
+
+    if (build_json_counter_key(json_key, ctx, entry) != NO_ERROR) {
+        json_decref(json_key);
+        return ENOMEM;
+    }
+    json_decref(json_key);
+
+    psabpf_counter_type_t type = psabpf_counter_get_type(ctx);
+    json_t *json_value = json_object();
+    if (json_value == NULL)
+        return ENOMEM;
+    json_object_set(parent, "value", json_value);
+
+    /* For counter values we cannot use built-in JSON integer type because
+     * it is signed type, but we need unsigned one.*/
+    if (type == PSABPF_COUNTER_TYPE_BYTES || type == PSABPF_COUNTER_TYPE_BYTES_AND_PACKETS) {
+        psabpf_counter_value_t bytes_value = psabpf_counter_entry_get_bytes(entry);
+        char *bytes_str = convert_bin_data_to_hexstr(&bytes_value, sizeof(psabpf_counter_value_t));
+        if (bytes_str != NULL) {
+            json_object_set_new(json_value, "bytes", json_string(bytes_str));
+            free(bytes_str);
+        } else {
+            json_decref(json_value);
+            return ENOMEM;
+        }
+    }
+    if (type == PSABPF_COUNTER_TYPE_PACKETS || type == PSABPF_COUNTER_TYPE_BYTES_AND_PACKETS) {
+        psabpf_counter_value_t packets_value = psabpf_counter_entry_get_packets(entry);
+        char *packets_str = convert_bin_data_to_hexstr(&packets_value, sizeof(psabpf_counter_value_t));
+        if (packets_str != NULL) {
+            json_object_set_new(json_value, "packets", json_string(packets_str));
+            free(packets_str);
+        } else{
+            json_decref(json_value);
+            return ENOMEM;
+        }
+    }
+
+    json_decref(json_value);
+    return NO_ERROR;
+}
+
+int do_counter_get(int argc, char **argv)
+{
+    int ret = EINVAL;
+    const char *counter_name = NULL;
+    psabpf_context_t psabpf_ctx;
+    psabpf_counter_context_t ctx;
+    psabpf_counter_entry_t entry;
+
+    psabpf_context_init(&psabpf_ctx);
+    psabpf_counter_ctx_init(&ctx);
+    psabpf_counter_entry_init(&entry);
+
+    if (parse_pipeline_id(&argc, &argv, &psabpf_ctx) != NO_ERROR)
+        goto clean_up_psabpf;
+
+    if (parse_dst_counter(&argc, &argv, &counter_name, &psabpf_ctx, &ctx) != NO_ERROR)
+        goto clean_up_psabpf;
+
+    bool counter_key_provided = false;
+    if (argc >= 1 && is_keyword(*argv, "key")) {
+        counter_key_provided = true;
+        if (parse_counter_key(&argc, &argv, &entry) != NO_ERROR)
+            goto clean_up_psabpf;
+    }
+
+    if (argc > 0) {
+        fprintf(stderr, "%s: unused argument\n", *argv);
+        goto clean_up_psabpf;
+    }
+
+    json_t *root = json_object();
```

Comment:
Could we separate this json stuff from `do_counter_get`? Maybe move this part to method like `get_json_counter_data` ?

Reply:
Done